planner/core/exhaust_physical_plans.go

```diff
@@ -547,14 +547,28 @@ func (p *LogicalJoin) buildRangeForIndexJoin(indexInfo *model.IndexInfo, innerPl
 		return nil, nil, nil
 	}
 
-	// We should guarantee that all the join's equal condition is used.
-	for _, eqCond := range eqConds {
+	// Guarantee res.AccessConds is not empty.
+	if len(res.AccessConds) == 0 {
```

Comment:
Is it possible that we fall into this branch?

Reply:
Yes, have a look at this sql:)
```sql
create table t(a int, b int, index idx_ab(a, b));
explain select * from t t1 join t t2 where t1.b = t2.b and t2.b is null
```